modules/vstudio/vs2005_dotnetbase.lua

```diff
@@ -838,7 +838,10 @@
 		-- do not remove the file generation for global.json otherwise MSTest will not work!!!
 		local globalpath = path.join(cfg.workspace.location, "global.json")
 		if cfg.dotnetsdk == "MSTest" and not os.isfile(globalpath) then
-			io.writefile(globalpath, '{"msbuild-sdks": {"MSTest.Sdk": "3.6.1"}}')
+			local content = p.capture(function() generate_global_json_content(prj) end)
+			if content ~= nil and #content > 0 then
+				p.generate(cfg.workspace, path.join(cfg.workspace.location,"global.json"), function() p.outln(content) end)
+			end
 		end
 	end
```

Comment:
I meant:

```suggestion
function dotnetbase.output_global_json(prj) -- unsure how you handle project with different config
		if cfg.dotnetsdk == "MSTest" then
			_p('{"msbuild-sdks": {"MSTest.Sdk": "3.6.1"}}')
    end
end

function dotnetbase.generate_global_json(prj)
			local content = p.capture(function() dotnetbase.output_global_json(prj) end)
			if content ~= nil and #content > 0 then
				p.generate(prj, path.join(prj.workspace.location,"global.json"), function() p.outln(content) end)
			end
end
```
In  vs2005.generateProject(prj)
Add `dotnetbase.generate_global_json(prj)` in `iscsharp` and in `isfsharp` branch.

In test, just check output of `dotnetbase.output_global_json`

Reply:
ill give it try!

Reply:
getting somewhere but now other test is broken

Reply:
fixed it it is now working